src/pg_to_evalscript/javascript_datacube/DataCube.js

```diff
@@ -269,20 +269,36 @@ class DataCube {
 
     extendFinalDimensionWithData(axis, dataToAdd) {
         const finalShape = this.getDataShape()
+        let loc = 1;
+        let elementsToInsertAtOnce = 1;
+        for (let i = 0; i < finalShape.length; i++) {
+            if (i < axis) {
+                elementsToInsertAtOnce *= finalShape[i];
+                continue
+            }
+            loc *= finalShape[i];
+        }
+        const finalData = this.insertIntoFinalDimension(dataToAdd, loc, dataToAdd.length / elementsToInsertAtOnce, axis);     
         finalShape[axis]++;
-        
-        const finalData = this.insertIntoFinalDimension(dataToAdd, finalShape[axis], finalShape[axis] - 1)     
         this.data = ndarray(finalData, finalShape)
     }
 
-    insertIntoFinalDimension(dataToAdd, dimensionSize, locationInDimension) {
+    insertIntoFinalDimension(dataToAdd, offset, elementsToInsertAtOnce, axis) {
```

Comment:
This method should be renamed as it not longer works only for the final dimension.